cli/src/commands/help.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::Write;
+
+use tracing::instrument;
+
+use crate::cli_util::CommandHelper;
+use crate::command_error::CommandError;
+use crate::ui::Ui;
+
+/// Print this message or the help of the given subcommand(s)
+#[derive(clap::Args, Clone, Debug)]
+pub(crate) struct HelpArgs {
+    /// The command to show help
+    command: Option<String>,
```

Comment:
The current clap-provided help supports things like `jj help git push`, so this probably need to be a `Vec<String>` to support that. Also add tests for that case and anything similar you can think of. Oh, `jj git help` and `jj git help push` also work, apparently. I don't know if users expect those combinations to work. It might be fine to ignore some of them. What do you and others think? Should we at least keep things like `jj git push help`? But if we do, it's probably not much harder to support `jj git help push` too.

Reply:
> The current clap-provided help supports things like `jj help git push`, so this probably need to be a `Vec<String>` to support that. Also add tests for that case and anything similar you can think of.

I agree, already working on that.

> Oh, `jj git help` and `jj git help push` also work, apparently. I don't know if users expect those combinations to work. It might be fine to ignore some of them. What do you and others think? Should we at least keep things like `jj git push help`? 
 
For me, `jj git help` is expected, but `jj git push help` not that much, but, as you said, if one is implemented, the other most probably will be too. I will think how to tackle this.

Reply:
> Should we at least keep things like `jj git push help`

Actually `jj git push help` doesn't work